File: WebAPI/PropertyHubAPI/PropertyHubAPI/DTO/Agencies/AgentCreateDto.cs
using PropertyHubAPI.DTO.Addresses;

namespace PropertyHubAPI.DTO.Agencies
{
    public class AgentCreateDto
    {
        public required string RegistrationNumber { get; set; }
        public required string Name { get; set; }
        public required string RegistrationCategory { get; set; }
        public required string BrokerageTradeName { get; set; }
        public required string BrokeragePhone { get; set; }
        public required string BrokerageEmail { get; set; }
        public required AddressDto BrokerageAddress { get; set; }
    }
}

File: client/src/models/AgentCreateDto.ts
import { AddressDto } from './AddressDto';

export interface AgentCreateDto {
    registrationNumber: string;
    name: string;
    registrationCategory: string;
    brokerageTradeName: string;
    brokeragePhone: string;
    brokerageEmail: string;
    brokerageAddress: AddressDto;
}